File: 128.LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums)
    {
        if (nums.empty()) {
            return 0;
        }
        
        sort(nums.begin(), nums.end());
        int largest = 1;
        int count = 1;
        
        for (size_t i = 1; i < nums.size(); ++i) {
            if (nums[i] == nums[i - 1] + 1) {
                count += 1;
                largest = max(largest, count);
            } else if (nums[i] != nums[i - 1]) { 
                count = 1;
            }
        }
        return largest;
    }
};

File: 128.LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        l = set(nums)
        longest = 0
        for num in l:
            if num - 1 not in l:
                current = 1
                while num + 1 in l:
                    current +=1
                    num+=1
                longest = max(current,longest)

        return longest
        